src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -0,0 +1,774 @@
+                this._cancellationTokenSourceRenewLeases.Cancel();
+                this._cancellationTokenSourceCheckForChanges.Dispose();
+                this._cancellationTokenSourceExecutor.Dispose();
+            }
+        }
+
+        /// <summary>
+        /// Queries the change/worker tables to check for new changes on the user's table. If any are found, stores the
+        /// change along with the corresponding data from the user table in "_rows".
+        /// </summary>
+        private async Task GetChangesAsync(CancellationToken token)
+        {
+            try
+            {
+                using (var connection = new SqlConnection(this._connectionString))
+                {
+                    await connection.OpenAsync(token);
+
+                    using (SqlTransaction transaction = connection.BeginTransaction(System.Data.IsolationLevel.RepeatableRead))
+                    {
+                        try
+                        {
+                            // Update the version number stored in the global state table if necessary before using it.
+                            using (SqlCommand updateTablesPreInvocationCommand = this.BuildUpdateTablesPreInvocation(connection, transaction))
+                            {
+                                await updateTablesPreInvocationCommand.ExecuteNonQueryAsync(token);
+                            }
+
+                            // Use the version number to query for new changes.
+                            using (SqlCommand getChangesCommand = this.BuildGetChangesCommand(connection, transaction))
+                            {
+                                var rows = new List<IReadOnlyDictionary<string, string>>();
+                                using (SqlDataReader reader = await getChangesCommand.ExecuteReaderAsync(token))
+                                {
+                                    while (await reader.ReadAsync(token))
+                                    {
+                                        rows.Add(SqlBindingUtilities.BuildDictionaryFromSqlRow(reader));
+                                    }
+                                }
+
+                                this._rows = rows;
+                            }
+
+                            // If changes were found, acquire leases on them.
+                            if (this._rows.Count > 0)
+                            {
+                                using (SqlCommand acquireLeasesCommand = this.BuildAcquireLeasesCommand(connection, transaction))
+                                {
+                                    await acquireLeasesCommand.ExecuteNonQueryAsync(token);
+                                }
+                            }
+                            transaction.Commit();
+                        }
+                        catch (Exception ex)
+                        {
+                            this._logger.LogError("Commit Exception Type: {0}", ex.GetType());
```

Comment:
Let's give this a better log message, something like `Exception getting list of changes for table...`